org.osgi.test.common/src/main/java/org/osgi/test/common/service/ServiceConfiguration.java

```diff
@@ -51,10 +52,18 @@ public ServiceConfiguration(ServiceConfigurationKey<S> key) {
 	public ServiceConfiguration(Class<S> serviceType, String format, String[] args, int cardinality, long timeout) {
 		this.serviceType = requireNonNull(serviceType);
 
-		Filter filter = format("(objectClass=%s)", serviceType.getName());
+		Filter filter = null;
 		format = String.format(requireNonNull(format), (Object[]) requireNonNull(args));
 		if (!format.isEmpty()) {
-			filter = format("(&%s%s)", filter.toString(), format);
+
+			if (AnyService.class.equals(serviceType)) {
+				filter = format(format);
+			} else {
+				filter = format("(objectClass=%s)", serviceType.getName());
+				filter = format("(&%s%s)", filter.toString(), format);
```

Comment:
This seems pointless to format a string and make it into a filter only to then toString the filter! Can't we just do one format call?

```java
filter = format("(&(objectClass=%s)%s)", serviceType.getName(), format);
```

Reply:
thank you for the hint